app/repositories/application/GeneralApplicationRepository.scala

```diff
@@ -319,47 +334,37 @@ class GeneralApplicationMongoRepository(timeZoneService: TimeZoneService)(implic
 
   override def submit(applicationId: String): Future[Unit] = {
     val query = BSONDocument("applicationId" -> applicationId)
-    val applicationStatusBSON = BSONDocument("$set" -> BSONDocument(
-      "applicationStatus" -> "SUBMITTED",
-      "progress-status.submitted" -> true
-    ))
-
-    collection.update(query, applicationStatusBSON, upsert = false) map {
-      case _ => ()
-    }
+    val updateBSON = BSONDocument("$set" -> applicationStatusBSON(Submitted))
+    collection.update(query, updateBSON, upsert = false) map { _ => }
   }
 
   override def withdraw(applicationId: String, reason: WithdrawApplicationRequest): Future[Unit] = {
     val query = BSONDocument("applicationId" -> applicationId)
-    val applicationStatusBSON = BSONDocument("$set" -> BSONDocument(
-      "withdraw" -> reason,
-      "applicationStatus" -> "WITHDRAWN",
-      "progress-status.withdrawn" -> true
-    ))
-
-    collection.update(query, applicationStatusBSON, upsert = false) map {
-      case _ => ()
-    }
+    val applicationBSON = BSONDocument("$set" -> BSONDocument(
+        "withdraw" -> reason
```

Comment:
Agreed. But this is a low level model change, I would rather defer this to withdraw story.